### PR TITLE
docs: Update the build docs for electron to improve doc readability

### PR DESCRIPTION
**Pull request by twitharshil**
In continuation of my efforts to improve doc for building electron, these are some improvements that I did which I feel improves the doc readability for someone who is new to building electron. 

**Release Notes**
Notes: none

### CODE REVIEW

File: docs/development/build-instructions-gn.md [hunk 1]
```diff
@@ -98,44 +98,57 @@ $ gclient sync -f
...
-To see the list of available build configuration options, run `gn args
-out/Testing --list`.
+**To generate Testing build config of Electron:**
 
-**For generating Testing build config of
-Electron:**
+On Linux & MacOS:
 
 ```sh
 $ gn gen out/Testing --args="import(\"//electron/build/args/testing.gn\") $GN_EXTRA_ARGS"
```
Comment (nornagon) [inline docs/development/build-instructions-gn.md:122]:
```suggestion
$ gn gen out/Testing --args="import(\"//electron/build/args/testing.gn\")"
```

I don't think GN_EXTRA_ARGS is relevant here; it's used in CI but never discussed in this document, so let's remove it.

File: docs/development/build-instructions-gn.md [hunk 2]
```diff
@@ -98,44 +98,57 @@ $ gclient sync -f
...
+```sh
+$ gn gen out/Testing --args="import(\"//electron/build/args/testing.gn\")"
+```
+
+**To generate Release (aka "non-component" or "static") build config of Electron:**
+
+On Linux & MacOS:
 
 ```sh
 $ gn gen out/Release --args="import(\"//electron/build/args/release.gn\") $GN_EXTRA_ARGS"
```
Comment (nornagon) [inline docs/development/build-instructions-gn.md:136]:
```suggestion
$ gn gen out/Release --args="import(\"//electron/build/args/release.gn\")"
```

File: docs/development/build-instructions-gn.md [hunk 3]
```diff
@@ -98,44 +98,57 @@ $ gclient sync -f
...
-Electron:**
+On Linux & MacOS:
 
 ```sh
 $ gn gen out/Testing --args="import(\"//electron/build/args/testing.gn\") $GN_EXTRA_ARGS"
 ```
 
-**For generating Release (aka "non-component" or "static") build config of
-Electron:**
+On Windows (without the optional argument):
```
Comment (nornagon) [inline docs/development/build-instructions-gn.md:125]:
```suggestion
On Windows:
```

File: docs/development/build-instructions-gn.md [hunk 4]
```diff
@@ -98,44 +98,57 @@ $ gclient sync -f
...
+
+```sh
+$ gn gen out/Testing --args="import(\"//electron/build/args/testing.gn\")"
+```
+
+**To generate Release (aka "non-component" or "static") build config of Electron:**
```
Comment (nornagon) [inline docs/development/build-instructions-gn.md:131]:
```suggestion
**To generate Release build config of Electron:**
```

We don't support component builds any more; all builds are statically linked.

File: docs/development/build-instructions-gn.md [hunk 5]
```diff
@@ -98,44 +98,57 @@ $ gclient sync -f
...
+
+On Linux & MacOS:
 
 ```sh
 $ gn gen out/Release --args="import(\"//electron/build/args/release.gn\") $GN_EXTRA_ARGS"
 ```
 
+On Windows (without the optional argument):
```
Comment (nornagon) [inline docs/development/build-instructions-gn.md:139]:
```suggestion
On Windows:
```

**Review by nornagon — APPROVED** [review]

File: docs/development/build-instructions-gn.md (hunk 1 above)
Comment (twitharshil) [reply docs/development/build-instructions-gn.md]: Done ✅

File: docs/development/build-instructions-gn.md (hunk 4 above)
Comment (twitharshil) [reply docs/development/build-instructions-gn.md]: Done ✅

**Comment by twitharshil** [comment]
@nornagon Any insights on why would `gclient sync` step would fail for the `docs-only` job?

**Comment by codebytere** [comment]
@twitharshil we've been having some issues there lately, it's not related to your pr :)

**Review by codebytere — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**